array_core/array_core.h

```diff
@@ -0,0 +1,109 @@
+#ifndef ARRAY_CORE_ARRAY_CORE_H_
+#define ARRAY_CORE_ARRAY_CORE_H_
+
+#include <iostream>
+#include <vector>
+#include "plotly_maker/plotly_maker.h"
+#include "utils_core.h"
+
+using std::vector;
+
+
+namespace davis {
+
+enum class visualizationTypes{
+    graph, heatmap, surface
+};
+
+enum class colorscales{
+    gray, rainbow
+};
+
+struct showSettings{
+    visualizationTypes visualType = visualizationTypes::heatmap;
+    colorscales colorscale = colorscales::rainbow;
+};
+
+
+//! two-dimensional vector
+template <typename T>
+bool show(const vector<vector<T>> &data);
+
+//! one-dimensional vector
+template <typename T>
+bool show(const vector<T> &data);
+
+//! two-dimensional array
+template <typename T>
+bool show(T **data, uint64_t arrRows, uint64_t arrCols);
+
+//! a one-dimensional array that simulates a two-dimensional one (element access [i*cols+j])
+template <typename T>
+bool show(const T *data, uint64_t arrRows, uint64_t arrCols);
+
+//! one-dimensional array
+template <typename T>
+bool show(const T *data);
+
+// *******************************
+// template functions realisations:
+// *******************************
+
+template <typename T>
+bool show(const vector<vector<T>> &data)
+{
+    vector<vector<double>> vecVecDbl;
+    vecVecDbl.reserve(data.size());
+    for (vector<T> row : data) {
+        vector<double> dblRow(row.begin(), row.end());
+        vecVecDbl.push_back(dblRow);
```

Comment:
https://ru.stackoverflow.com/questions/570042/%D0%A0%D0%B0%D0%B7%D0%BD%D0%B8%D1%86%D0%B0-%D0%BC%D0%B5%D0%B6%D0%B4%D1%83-push-back-%D0%B8-emplace-back

https://pro-prof.com/forums/topic/%D1%8D%D1%84%D1%84%D0%B5%D0%BA%D1%82%D0%B8%D0%B2%D0%BD%D0%BE%D0%B5-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-vector-%D0%B2-c

Maybe emplace_back is better to use